test/functional/p2p_peer_logging.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test node peer logging capability.
+
+Additionally, the output of contrib/peer-logging/peer-logging-parser.py should be verified manually.
+"""
+
+import glob
+from io import BytesIO
+import os
+
+from test_framework.mininode import P2PDataStore
+from test_framework.test_framework import BitcoinTestFramework
+
+TIME_SIZE = 8
+LENGTH_SIZE = 4
+COMMAND_SIZE = 12
+
+def mini_parser(dat_file):
+    """Parse a data file created by LogMessage.
+
+    From the data file we'll only check the structure.
+
+    We won't care about things like:
+    - Deserializing the payload of the message
+        - This is managed by the deserialize methods in test_framework.messages
+    - The order of the messages
+        - There's no reason why we can't, say, change the order of the messages in the handshake
+    - Message Type
+        - We can add new message types
+
+    We're ignoring these because they're simply too brittle to test here.
+    """
+    with open(dat_file, 'rb') as f_in:
+        while True:
+            tmp_header_raw = f_in.read(TIME_SIZE + LENGTH_SIZE + COMMAND_SIZE)
+            if not tmp_header_raw:
+                break
+            tmp_header = BytesIO(tmp_header_raw)
+            time = int.from_bytes(tmp_header.read(TIME_SIZE), "little") # type: int
+            assert(time >= 1231006505000000)   # genesis block timestamp
+            command = tmp_header.read(COMMAND_SIZE).split(b'\x00', 1)[0]  # type: bytes
+            assert(len(command) > 0)
+            length = int.from_bytes(tmp_header.read(LENGTH_SIZE), "little") # type: int
+            data = f_in.read(length)
+            assert(len(data) == length)
```

Comment:
would be good to use assert_equal for debugability 

https://travis-ci.org/github/bitcoin/bitcoin/jobs/712616300#L3056

Reply:
done